docs/src/ref/gfi.md

```diff
@@ -81,54 +93,61 @@ There is also a family of probability distributions ``q(r; x, t)`` on untraced r
 q(r; x, t) > 0 \mbox{ if and only if } p(r; x, t) > 0
 ```
 
+
 ## Traces
 
 An **execution trace** (or just *trace*) is a record of an execution of a generative function.
-There is no abstract type representing all traces.
-Different concrete types of generative functions use different data structures and different Julia types for their traces.
-The trace type that a generative function uses is the second type parameter of the [`GenerativeFunction`](@ref) abstract type.
+Traces are the primary data structures manipulated by Gen inference programs.
+There are various methods for producing, updating, and inspecting traces.
+Traces contain:
 
-A generative function can be executed and traced using:
-```@docs
-simulate
-```
+- the arguments to the generative function
 
-An traced execution that satisfies constraints on the choice map can be generated using:
-```@docs
-generate
-```
+- the random choice map 
 
-The trace contains various information about the execution, including:
+- the return value
 
-The arguments to the generative function:
-```@docs
-get_args
-```
+- auxiliary state
 
-The return value of the generative function:
-```@docs
-get_retval
-```
+- other implementation-specific state that is not exposed to the caller or user of the generative function, but is used internally to facilitate e.g. incremental updates to executions.
 
-The map ``t`` from addresses of random choices to their values:
+Different concrete types of generative functions use different data structures and different Julia types for their traces, but traces are subtypes of [`Trace`](@ref).
 ```@docs
-get_choices
+Trace
 ```
+The concrete trace type that a generative function uses is the second type parameter of the [`GenerativeFunction`](@ref) abstract type.
+For example, the trace type of [`DynamicDSLFunction`](@ref) is `DynamicDSLTrace`.
 
-The log probability that the random choices took the values they did:
-```@docs
-get_score
+A generative function can be executed to produce a trace of the execution using [`simulate`](@ref):
+```julia
+trace = simulate(foo, (a, b))
+```
+An traced execution that satisfies constraints on the choice map can be generated using [`generate`](@ref):
+```julia
+trace, weight = generate(foo, (a, b), choicemap((:z, false)))
 ```
+There are various methods for inspecting traces, including:
 
-A reference to the generative function that was executed:
-```@docs
-get_gen_fn
+- [`get_args`](@ref) (returns the arguments to the function)
+
+- [`get_retval`](@ref) (returns the return value of the function)
+
+- [`get_choices`](@ref) (returns the choice map)
```

Comment:
Do we also want a `get_aux` (and/or something that returns a combined `merge(choice_map, aux)`)?  `getindex` might be enough (whereas I do see why we need a `get_choices`), just wondering what you think.

Reply:
I don't think we need that.